test/common/protobuf/utility_test.cc

```diff
@@ -1720,6 +1721,36 @@ TEST_F(DeprecatedFieldsTest, DEPRECATED_FEATURE_TEST(IndividualFieldDeprecatedEm
   EXPECT_EQ(1, deprecated_feature_seen_since_process_start_.value());
 }
 
+// Make sure recursing into an Any works if asked.
+TEST_F(DeprecatedFieldsTest, DEPRECATED_FEATURE_TEST(RecurseIntoAny)) {
+  envoy::service::discovery::v3::DiscoveryResponse discovery_response_config;
+  TestUtility::loadFromFile(
+      TestEnvironment::runfilesPath("test/tools/schema_validator/test/config/lds_deprecated.yaml"),
```

Comment:
I'm on the fence, but not relying on complex out-of-directory test cases seems generally better, e.g. if we can create small examples locally that provide coverage.

Reply:
The only reason I added these tests is that for reasons I don't understand the shell test does not seem to count towards coverage. Maybe this won't be an issue with a gtest test, and then I can just remove this duplication. So I guess that would be points in favor of removing the shell test.